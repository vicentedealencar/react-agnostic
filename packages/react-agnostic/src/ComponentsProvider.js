import React from 'react'
import { Provider } from './componentsContext'
import defaults from './defaults'

export default class ComponentsProvider extends React.PureComponent {
  state = {
    components: {
      ...defaults,
      ...this.props.components,
    },
  }

  render() {
    const { children } = this.props
    const { components } = this.state

    if (!Provider) {
      throw new Error('Provider not defined. Try using react > 16')
    }

    return <Provider value={components}>{children}</Provider>
  }
}
