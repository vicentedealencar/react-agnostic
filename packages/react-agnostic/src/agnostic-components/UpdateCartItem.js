import React from 'react'
import withComponents from '../withComponents'

const UpdateCartItem = ({item, updateCartItem, components}) => {
  const {Button, Text, View} = components

  return updateCartItem
    ? React.createElement(
        View,
        null,
        React.createElement(
          Button,
          {
            disabled: item.amount <= 0,
            onClick: function onClick() {
              return updateCartItem(
                Object.assign({}, item, {amount: item.amount - 1}),
              )
            },
          },
          '-',
        ),
        React.createElement(Text, null, item.amount),
        React.createElement(
          Button,
          {
            onClick: function onClick() {
              return updateCartItem(
                Object.assign({}, item, {amount: item.amount + 1}),
              )
            },
          },
          '+',
        ),
      )
    : React.createElement(Text, null, item.amount)
  // return updateCartItem ? (
  //   <View>
  //     <Button
  //       disabled={item.amount <= 0}
  //       onClick={() => updateCartItem({...item, amount: item.amount - 1})}
  //     >
  //       -
  //     </Button>
  //     <Text>{item.amount}</Text>
  //     <Button
  //       onClick={() => updateCartItem({...item, amount: item.amount + 1})}
  //     >
  //       +
  //     </Button>
  //   </View>
  // ) : (
  //   <Text>{item.amount}</Text>
  // )
}

export default withComponents(UpdateCartItem)
