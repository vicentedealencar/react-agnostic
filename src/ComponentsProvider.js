import React from 'react'
import {Provider} from './componentsContext'
import defaults from './defaults'

export default class ComponentsProvider extends React.PureComponent {
  state = {
    components: {
      ...defaults,
      ...this.props.components,
    },
  }

  render() {
    const {children} = this.props
    const {components} = this.state

    return <Provider value={components}>{children}</Provider>
  }
}
