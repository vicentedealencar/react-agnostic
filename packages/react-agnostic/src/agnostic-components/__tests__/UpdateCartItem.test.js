import React from 'react'
import renderer from 'react-test-renderer'
import UpdateCartItem from '../UpdateCartItem'
import ComponentsProvider from '../../ComponentsProvider'

test('UpdateCartItem renders correctly', () => {
  const component = <UpdateCartItem
    updateCartItem={() => {}}
    item={{
      name: 'ball',
      amount: 3,
      price: 2,
    }} />
  const tree = renderer
    .create(component)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('UpdateCartItem renders with components passed by props', () => {
  const View = 'section'
  const Text = 'p'
  const Button = 'a'
  const components = {
    View,
    Text,
    Button,
  }

  const component = <UpdateCartItem
    components={components}
    updateCartItem={() => {}}
    item={{
      name: 'ball',
      amount: 3,
      price: 2,
    }} />
  const tree = renderer
    .create(component)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('UpdateCartItem renders with components passed by Provider', () => {
  const View = 'section'
  const Text = 'p'
  const Button = 'a'
  const components = {
    View,
    Text,
    Button,
  }

  const component = <ComponentsProvider components={components}>
    <UpdateCartItem
      updateCartItem={() => {}}
      item={{
        name: 'ball',
        amount: 3,
        price: 2,
      }} />
  </ComponentsProvider>
  const tree = renderer
    .create(component)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
