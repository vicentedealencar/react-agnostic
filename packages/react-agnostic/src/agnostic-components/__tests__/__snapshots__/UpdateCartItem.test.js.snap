// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateCartItem renders correctly 1`] = `
<div>
  <button
    disabled={false}
    onClick={[Function]}
  >
    -
  </button>
  <span>
    3
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`UpdateCartItem renders with components passed by Provider 1`] = `
<section>
  <a
    disabled={false}
    onClick={[Function]}
  >
    -
  </a>
  <p>
    3
  </p>
  <a
    onClick={[Function]}
  >
    +
  </a>
</section>
`;

exports[`UpdateCartItem renders with components passed by props 1`] = `
<section>
  <a
    disabled={false}
    onClick={[Function]}
  >
    -
  </a>
  <p>
    3
  </p>
  <a
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
